addon/components/lazy-render-wrapper.js

```diff
@@ -206,6 +206,7 @@ export default Ember.Component.extend({
 				} else {
 					this.set('_hasUserInteracted', true);
 					this.set('_shouldShowOnRender', true);
+					this.set('_isInProcessOfShowing', true);
```

Comment:
I think that the use of private variables is the best approach when used within a single file. But when passed from component to component we have to pass them, as you've identified.